.gitlab/build-lib-init.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+set -e
+
+# Safety checks to make sure we have required values
+if [ -z "$CI_COMMIT_TAG" ]; then
+  echo "Error: CI_COMMIT_TAG was not provided"
+  exit 1
+fi
+
+if [ -z "$CI_COMMIT_SHA" ]; then
+  echo "Error: CI_COMMIT_SHA was not provided"
+  exit 1
+fi
+
+if [ -z "$IMG_DESTINATION_BASE" ]; then
+  echo "Error: IMG_DESTINATION_BASE. This should be set to the destination docker image, excluding the tag name, e.g. dd-lib-dotnet-init"
+  exit 1
+fi
+
+# If this is a pre-release release, we don't publish
+if echo "$CI_COMMIT_TAG" | grep -q "-" > /dev/null; then
+  echo "Error: This is a pre-release version, should not publish images: $CI_COMMIT_TAG"
+  exit 1
+fi
+
+# Calculate the tags we use for floating major and minor versions
+MAJOR_MINOR_VERSION="$(sed -nE 's/^(v[0-9]+\.[0-9]+)\.[0-9]+$/\1/p' <<< ${CI_COMMIT_TAG})"
+MAJOR_VERSION="$(sed -nE 's/^(v[0-9]+)\.[0-9]+\.[0-9]+$/\1/p' <<< ${CI_COMMIT_TAG})"
+
+# Make sure we have all the tags
+git fetch --tags
+
+# We need to determine whether this is is the latest tag and whether it's the latest major or not
+# So we fetch all tags and sort them to find both the latest, and the latest in this major.
+# 'sort' technically gets prerelease versions in the wrong order here, but we explicitly
+# exclude them anyway, as they're ignored for the purposes of determining the 'latest' tags.
+LATEST_TAG="$(git tag | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V -r | head -n 1)"
+LATEST_MAJOR_TAG="$(git tag -l "$MAJOR_VERSION.*" | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V -r | head -n 1)"
```

Comment:
Actually, we can make these even more restrictive

```suggestion
LATEST_TAG="$(git tag | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V -r | head -n 1)"
LATEST_MAJOR_TAG="$(git tag -l "$MAJOR_VERSION.*" | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V -r | head -n 1)"
```